ci-operator/step-registry/rosa/nodes/wait-ready/rosa-nodes-wait-ready-commands.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+CLUSTER_ID=$(cat "${SHARED_DIR}/cluster-id")
+echo "CLUSTER_ID is $CLUSTER_ID"
+
+# Configure aws
+CLOUD_PROVIDER_REGION=${LEASED_RESOURCE}
+AWSCRED="${CLUSTER_PROFILE_DIR}/.awscred"
+if [[ -f "${AWSCRED}" ]]; then
+  export AWS_SHARED_CREDENTIALS_FILE="${AWSCRED}"
+  export AWS_DEFAULT_REGION="${CLOUD_PROVIDER_REGION}"
+else
+  echo "Did not find compatible cloud provider cluster_profile"
+  exit 1
+fi
+
+# Log in
+ROSA_VERSION=$(rosa version)
+ROSA_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/ocm-token")
+if [[ ! -z "${ROSA_TOKEN}" ]]; then
+  echo "Logging into ${ROSA_LOGIN_ENV} with offline token using rosa cli ${ROSA_VERSION}"
+  rosa login --env "${ROSA_LOGIN_ENV}" --token "${ROSA_TOKEN}"
+  if [ $? -ne 0 ]; then
+    echo "Login failed"
+    exit 1
+  fi
+else
+  echo "Cannot login! You need to specify the offline token ROSA_TOKEN!"
+  exit 1
+fi
+
+# Get desired node count
+compute_count=$(rosa describe cluster -c "$CLUSTER_ID"  -o json  |jq -r '.nodes.compute')
+if [ "$compute_count" = "null" ]; then 
+  echo "--auto-scaling enabled, retrieving min_replicas count desired"
+  desired_node_count=$(rosa describe cluster -c "$CLUSTER_ID" -o json  | jq -r '.nodes.autoscale_compute.min_replicas')
+else
+  echo "--replicas set, retrieving replica count desired"
+  desired_node_count=$(rosa describe cluster -c "$CLUSTER_ID"  -o json  |jq -r '.nodes.compute')
+fi
+
+echo "Desired worker node count: $desired_node_count"
+
+# wait for all nodes to reach Ready=true to ensure that all machines and nodes came up, before we run
+# any e2e tests that might require specific workload capacity.
+echo "$(date) - waiting for worker nodes to be ready..."
+ret=0
+oc wait nodes --all --for=condition=Ready=true --timeout=10m || ret=$?
```

Comment:
I assume `rosa-nodes-wait-ready` should be the requried step for the test. Based on this, we do not need to add the pipe to make sure this step must pass. If this step fails, the whole workflow fails. This depends on your OCP experience.

Besides, if there are a large workers, the `timeout=10m` may be not enough, 30m could be better.  This is based on your performance testing experience.

Reply:
@svetsa-rh I agree with Yanming. As this is added to the 'rosa-aws-sts-provision' chain, a user of this chain can configure what ever REPLICAS env they want. Setting timeout to a bigger value, or set it based on the replica number will be safter. Adding a log to tell user the timeout setting is also helpful.
And because we can't have the cluster wait for debug (deprovision will be run when previous step fails), if we can add more debug info of the machine/node that is not ready within the timeout, it would be easier for the user to know what was happening.

Reply:
Thx for the feedback! Will update.